scripts/architecture_lint/compatibility.py

```
@@ -0,0 +1,143 @@
+import re
+from pathlib import Path
+
+from .baseline import load_json_file
+from .model import Diagnostic
+from .repository import read_text
+
+
+RULE_ID = "COMPAT-LEDGER"
+_SEMVER_NUMBER = r"(?:0|[1-9][0-9]*)"
+_SEMVER_PRERELEASE_IDENTIFIER = (
+    rf"(?:0|[1-9][0-9]*|[0-9A-Za-z-]*[A-Za-z-][0-9A-Za-z-]*)"
```

Ruff (0.16.0)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)